### ISSUE TITLE
Overload typings with Python *args in super class

### ISSUE THREAD

**Issue by JKillian** [python/mypy#6907]
_Are you reporting a bug, or opening a feature request?_
A bug

_Please insert below the code you are checking with mypy,_

```py
from typing import overload, Any
from typing_extensions import Literal

class Base:
  def get(self, *names: str) -> Any:
    pass

class Sub(Base):
  @overload # error: Signature of "get" incompatible with supertype "Base"
  def get(self, *names: Literal['Something']) -> int: ...
  @overload
  def get(self, *names: Literal['Else']) -> str: ...
  def get(self, *names: str) -> Any:
    return super().get(*names)
```

_What is the actual behavior/output?_
See comment in code above: `myfile.py:10: error: Signature of "get" incompatible with supertype "Base"`

_What is the behavior/output you expect?_
No error

_What are the versions of mypy and Python you are using?_
Python 3.6.5
Mypy 0.701
typing-extensions 3.7.2

In short, I'm trying to figure out how to provide more specific types in a subclass when overriding a base class method that takes `*args`. I tried several incantations of overloads, but unfortunately none of them seemed to work. I don't believe the error is correct in this case, as far as I can tell, my overload signatures are compatible with the super class's types.

(Full disclaimer, I did [ask this same question over in SO](https://stackoverflow.com/q/56350385/3124288). If anyone provides an answer here, happy to also accept their answer in SO!)

**Comment by JelleZijlstra** [comment]
Your code is unsafe because a call like `Base().get("some random string")` would be accepted by the type checker, but `Sub().get("some random string")` would not.

I'm also not sure what you're trying to achieve here (unless this is minimized from other code?). The subclass would only allow calls like `Sub().get("Something", "Something", "Something")`, for any number of arguments, and I'm having a hard time imagining a situation where that is useful. Perhaps you're misunderstanding how `*args` typing or Literals work.

**Comment by JKillian** [comment]
> Your code is unsafe because a call like Base().get("some random string") would be accepted by the type checker, but Sub().get("some random string") would not.

Ah sorry, this was due to a slight misunderstanding on my part of how `overload` works. My initial understanding was that if the implementation types where considered another possible case of types for the function, however, after playing around with things, I see that's not the case. So if I define a third overload with the same signature as the implementation, things work fairly well:"

```py
class Sub(Base):
  @overload
  def get(self, *names: Literal['Something']) -> int: ...
  @overload
  def get(self, *names: Literal['Else']) -> str: ...
  @overload
  def get(self, *names: str) -> Any:  # define this overload case to be the same as the implementation
  def get(self, *names: str) -> Any:
    return super().get(*names)
```

> I'm also not sure what you're trying to achieve here (unless this is minimized from other code?). The subclass would only allow calls like `Sub().get("Something", "Something", "Something"`)

Agreed that this is silly. I was only typing things this was because I couldn't figure out how to make things work for the single argument case. However, with the realization noted at the top of this comment, it's easy enough to make things work in a more ideal way now, using mypy's [positional-only parameters](https://mypy.readthedocs.io/en/latest/cheat_sheet_py3.html#functions) (until true positional-only params come out!):

```py
class Sub(Base):
  @overload
  def get(self, __name: Literal['Something']) -> int: ...
  @overload
  def get(self, __name: Literal['Else']) -> str: ...
  @overload
  def get(self, *names: str) -> Any:  # define this overload case to be the same as the implementation
  def get(self, *names: str) -> Any:
    return super().get(*names)
```

I'm going to close this issue out since the types in the last code-snipper above work great for my usecase. Sorry for the support-like ticket here, thanks for the help @JelleZijlstra!
